cluster-up/cluster/kind/common.sh

```diff
@@ -141,6 +141,28 @@ function _fix_node_labels() {
     done
 }
 
+function _get_cri_bridge_mtu() {
+  docker network inspect bridge  | grep mtu | awk '{print $2}' | sed "s/\"//g"
+}
+
+function _patch_calico_manifest_mtu() {
```

Comment:
It is not common to mutate config file in-place.
I suggest you change the original file to `kube-calico.yaml.in` and place instead of the mtu a placeholder `@CNI_MTU@`, that you later can `sed` and create a `kube-calico.yaml` which you can apply.

Reply:
consider please envsubst if you put a place holder

Reply:
We can do the same we did for qemu based providers, use diff file and a manifest w/o any change

Reply:
this we done for calico
but for kubeadm we used envsubst
cluster-provision/k8s/1.18/provision.sh:envsubst < /tmp/kubeadm.conf > $kubeadm_manifest